uninstall.sh

```diff
@@ -1,31 +1,25 @@
 #!/usr/bin/env bash
-if [ -z "$BASH_IT" ];
-then
-  BASH_IT="$HOME/.bash_it"
-fi
+: "${BASH_IT:=${HOME?}/.bash_it}"
 
-case $OSTYPE in
-  darwin*)
-    CONFIG_FILE=.bash_profile
-    ;;
-  *)
-    CONFIG_FILE=.bashrc
-    ;;
-esac
+CONFIG_FILE=".bashrc"
+if [[ ! -e ~/.bashrc && -e ~/.bash_profile ]]; then
+	# legacy Mac or WSL or just no backup file
```

Comment:
you can check for the backup first to determine which one is the correct one

Reply:
The weird OS-dependent logic from before left so so so many edge-cases. Here, for example, there's no way to tell if the user simply doesn't have a customized `~/.bashrc` on purpose or what. I imagine that someone who thoroughly knows the difference may not use the install scripts, though, so maybe best-guess is good enough...

And if the user has *removed* `~/.bashrc` then we don't want to re-create one by restoring a backup file. The backup should *restore* what was changed by the installer, but should avoid anything that would be significantly not-that.

I'll see what the brain worms can come up with 😃

Reply:
@gaelicWizard In mac os, there is a problem with the bash profile as during installation it had created initialization in bash_profile and when I installed the bashit it picked bash profile file. Now how do you know whether or not bashrc is sourced in the bash_profile for login

Reply:
@tbhaxor, yes that's exactly the bit I'm worrying about.

Reply:
what do you think @gaelicWizard?
I am worried about breaking changes here. Some folks just use bash-it as instructed, and they have bash-it installation in their .bash_profile file on mac. We need to make sure that in installation we simply remove our startup from the user's startup scripts and not do any funny business

Reply:
Ok, I've just pushed a version that attempts to detect which file we installed with by `fgrep`ing for `BASH_IT`. If someone installed with `install.sh`, then the file definitely would include `BASH_IT`. 

There's three failure conditions:
1. no config files nor backup files exist. This is unlikely.
2. Both config files exist *and* match. This is unlikely, but if it happens we should just bail. The user obviously did something and they should fix it.
3. Neither config file matches `BASH_IT`. This indicates that _Bash It_ was never installed, *or* that the user uses some other loader (like `bashrc_dispatch`); either way, the config files aren't from `install.sh` so we don't touch them.

Success conditions are:
1. `~/.bashrc` matches `BASH_IT`, so that's our file.
2. `~/.bash_profile` matches, so that's our file.

What have I missed?